xarray/plot/plot.py

```diff
@@ -555,14 +563,14 @@ def _infer_interval_breaks(coord):
 
 
 @_plot2d
-def pcolormesh(x, y, z, ax, **kwargs):
+def pcolormesh(x, y, z, ax, infer_intervals=True, **kwargs):
     """
     Pseudocolor plot of 2d DataArray
 
     Wraps matplotlib.pyplot.pcolormesh
     """
 
-    if not hasattr(ax, 'projection'):
+    if infer_intervals:
```

Comment:
I'm not sure to understand... ``infer_intervals=True`` was the *de-facto* default until V0.7.2, and from the example above it is clear that setting it to ``False`` is almost never what you want to do. Can you provide an example where this is a good thing?

I agree your concerns about breaking old code though. How about keeping the current behavior for one more version, but issuing a warning that in the next version (probably 0.9.1 or 0.10), it is going to be set to True per default? 

For my purposes this is going to imply that I'm going to use ``imshow()`` and not ``pcolormesh`` anymore for the time being.